monitoring/harmony-monitor/blockchain-watchdog/reporting-server.go

```diff
@@ -294,91 +315,162 @@ type headerInfoRPCResult struct {
 	Payload headerInformation
 	IP      string
 }
+
 type noReply struct {
 	IP            string
 	FailureReason string
 	RPCPayload    string
 }
 
+type MetadataContainer struct {
+	TS    time.Time
+	Nodes []metadataRPCResult
+	Down  []noReply
+}
+
+type BlockHeaderContainer struct {
+	TS    time.Time
+	Nodes []headerInfoRPCResult
+	Down  []noReply
+}
+
 type monitor struct {
-	chain            string
-	MetadataSnapshot struct {
-		TS    time.Time
-		Nodes []metadataRPCResult
-	}
-	BlockHeaderSnapshot struct {
-		TS    time.Time
-		Nodes []headerInfoRPCResult
+	chain               string
+	inUse               sync.WaitGroup
+	WorkingMetadata     MetadataContainer
+	WorkingBlockHeader  BlockHeaderContainer
+	MetadataSnapshot    MetadataContainer
+	BlockHeaderSnapshot BlockHeaderContainer
+	SummarySnapshot     map[string]map[string]interface{}
+	NoReplySnapshot     []noReply
+	consensusProgress   map[string]bool
+}
+
+type work struct {
+	address string
+	rpc     string
+	body    []byte
+}
+
+type reply struct {
+	address    string
+	rpc        string
+	rpcPayload []byte
+	rpcResult  []byte
+	oops       error
+}
+
+func (m* monitor) worker(jobs chan work, channels map[string](chan reply), groups map[string]*sync.WaitGroup) {
+	for j := range jobs {
+		result := reply{address : j.address, rpc : j.rpc}
+		result.rpcResult, result.rpcPayload, result.oops = request(
+			"http://" + j.address, j.body)
+		channels[j.rpc] <- result
+		groups[j.rpc].Done()
 	}
-	SummarySnapshot   map[string]map[string]interface{}
-	NoReplyMachines   []noReply
-	lock              *sync.Mutex
-	cond              *sync.Cond
-	consensusProgress map[string]bool
 }
 
-func (m *monitor) update(
-	rpc string, every int, superCommittee map[int]committee,
-) {
-	type t struct {
-		addr       string
-		rpcPayload []byte
-		rpcResult  []byte
-		oops       error
+func (m* monitor) manager(jobs chan work, interval int, nodeList []string,
+					rpc string, group *sync.WaitGroup, channels map[string](chan reply)) {
+	for now := range time.Tick(time.Duration(interval) * time.Second) {
+		queryID := 0
+		group.Add(len(nodeList))
+		for _, n := range nodeList {
+			requestBody, _ := json.Marshal(map[string]interface{} {
```

Comment:
don't want to make majority of this data over & over. Make it once for each RPC, then just update the key "id"'s value